### PR TITLE
[Bootcamp] InfoRequest model

### PR DESCRIPTION
**Pull request by dplamenov**
Create CRUD for **InfoRequest** model.
These are new paths: 
info-request

### CODE REVIEW

File: apps/api/src/info-request/info-request.service.ts [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+
+@Injectable()
+export class InfoRequestService {
+  constructor(public prisma: PrismaService){}
+
+  create(createInfoRequestDto: CreateInfoRequestDto) {
+    return this.prisma.infoRequest.create({data: createInfoRequestDto});
+  }
+
+  findAll() {
```
Comment (kachar) [inline apps/api/src/info-request/info-request.service.ts:14]: @dplamenov  At the moment partially this service has been done in `SupportService`

https://github.com/podkrepi-bg/api/blob/b3ec755bb400e55715400080e3c6306d4c306a5f/apps/api/src/support/support.service.ts#L40-L56

It's kind of duplication as the creation of info request would be expected to do the same thing and at the moment it does two separate things.

Here are some ideas how to go further:

1. Remove the creation methods from `InfoRequestService`
2. Move the logic from `SupportService` to `InfoRequest` service and make them dependent
3. Something else that you might think of


Let's discuss the further steps to refactor this scenario so the code is located only on one place

File: apps/api/src/info-request/info-request.controller.ts [hunk 2]
```diff
@@ -0,0 +1,40 @@
...
+  }
+
+  @Get(':id')
+  @Public()
+  findOne(@Param('id') id: string) {
+    return this.infoRequestService.findOne(id);
+  }
+
+  @Patch(':id')
+  @Public()
```
Comment (kachar) [inline apps/api/src/info-request/info-request.controller.ts:30]: Let's make all editorial endpoints actions private, so noone is able to edit a info request without permissions

```suggestion
```

File: apps/api/src/info-request/info-request.controller.ts [hunk 3]
```diff
@@ -0,0 +1,40 @@
...
+  update(@Param('id') id: string, @Body() updateInfoRequestDto: UpdateInfoRequestDto) {
+    return this.infoRequestService.update(id, updateInfoRequestDto);
+  }
+
+  @Delete(':id')
+  @Public()
```
Comment (kachar) [inline apps/api/src/info-request/info-request.controller.ts:36]:
```suggestion
```

File: apps/api/src/info-request/info-request.controller.ts [hunk 4]
```diff
@@ -0,0 +1,40 @@
...
+  }
+
+  @Get('list')
+  @Public()
+  findAll() {
+    return this.infoRequestService.findAll();
+  }
+
+  @Get(':id')
+  @Public()
```
Comment (kachar) [inline apps/api/src/info-request/info-request.controller.ts:24]:
```suggestion
```

File: apps/api/src/info-request/dto/create-info-request.dto.ts [hunk 5]
```diff
@@ -0,0 +1,14 @@
+import { ApiProperty } from '@nestjs/swagger'
+import { Expose } from 'class-transformer'
+import { IsString } from 'class-validator'
+
+export class CreateInfoRequestDto {
```
Comment (kachar) [inline apps/api/src/info-request/dto/create-info-request.dto.ts:5]: This class actually copies the functionality of `CreateInquiryDto`

https://github.com/podkrepi-bg/api/blob/c4355b6f8221dc7a0e80ab235c55e142f3493174/apps/api/src/support/dto/create-inquiry.dto.ts#L8

Let's leave them co-exsist for now and will check later on for refactoring

**Review by kachar — CHANGES_REQUESTED** [review]
@dpetrov Let's discuss how to go further with the refactoring of this PR as it copies already working functionallity

File: apps/api/src/info-request/info-request.service.ts (hunk 1 above)
Comment (dplamenov) [reply apps/api/src/info-request/info-request.service.ts:14]: Can I delete the InfoRequestService and use SupportService at InfoRequestController?

Comment (dplamenov) [reply apps/api/src/info-request/info-request.service.ts:15]: I committed changes but I am not sure is already ok.

**Review by kachar — APPROVED** [review]

File: apps/api/src/info-request/info-request.service.ts [hunk 6]
```diff
@@ -0,0 +1,30 @@
+import { Injectable } from '@nestjs/common';
+import { PrismaService } from '../prisma/prisma.service';
+import { CreateInquiryDto } from '../support/dto/create-inquiry.dto';
+import { SupportService } from '../support/support.service';
+import { UpdateInfoRequestDto } from './dto/update-info-request.dto';
+
+@Injectable()
+export class InfoRequestService {
+  constructor(public prisma: PrismaService, public supportService: SupportService){}
```
Comment (kachar) [inline apps/api/src/info-request/info-request.service.ts:9]:
```suggestion
  constructor(private prisma: PrismaService, private supportService: SupportService){}
```

File: apps/api/src/info-request/info-request.controller.ts [hunk 7]
```diff
@@ -0,0 +1,37 @@
...
+  constructor(private readonly infoRequestService: InfoRequestService) {}
+
+  @Post()
+  @Public()
+  create(@Body() createInfoRequestDto: CreateInquiryDto) {
+    return this.infoRequestService.create(createInfoRequestDto);
+  }
+
+  @Get('list')
+  @Public()
```
Comment (kachar) [inline apps/api/src/info-request/info-request.controller.ts:18]:
```suggestion
```

**Pull request merged**